biggraphite/drivers/elasticsearch.py

```diff
@@ -600,29 +600,32 @@ def get_metric(self, metric_name, touch=False):
 
         metric_name = bg_accessor.sanitize_metric_name(metric_name)
 
-        metric = self.__get_metric(metric_name)
-        if metric is None:
+        document = self.__get_document(metric_name)
+        if document is None:
             return None
 
         if touch:
-            self.__touch_metadata_on_need(metric)
+            self.__touch_metadata_on_need(document)
+
+        return self._document_to_metric(document)
 
+    def _document_to_metric(self, document):
         metadata = bg_accessor.MetricMetadata.from_string_dict(
-            metric.config.to_dict()
+            document.config.to_dict()
         )
         return self.make_metric(
-            metric_name,
+            document.name,
             metadata,
-            created_on=str_to_datetime(metric.created_on),
-            updated_on=str_to_datetime(metric.updated_on),
-            read_on=str_to_datetime(metric.read_on)
+            created_on=str_to_datetime(document.created_on),
```

Comment:
I don't remember why these are strings, do you know ? can we put a comment explaining why ?

Reply:
I don't, I'll take a look why

Reply:
The search API of the dsl get strings. We could use other APIs from the dsl. Added a comment in the code